osu.Game.Tests/Visual/Gameplay/TestSceneScoring.cs

```diff
@@ -0,0 +1,498 @@
+                                    }
+                                },
+                            },
+                        }
+                    }
+                };
+
+                sliderMaxCombo.Current.BindValueChanged(_ => rerun());
+
+                graphs.MissLocations.BindCollectionChanged((_, __) => rerun());
+                graphs.NonPerfectLocations.BindCollectionChanged((_, __) => rerun());
+
+                graphs.MaxCombo.BindTo(sliderMaxCombo.Current);
+
+                rerun();
+            });
+        }
+
+        private const int base_great = 300;
+        private const int base_ok = 100;
+
+        private void rerun()
+        {
+            graphs.Clear();
+            legend.Clear();
+
+            runForProcessor("lazer-standardised", Color4.YellowGreen, new ScoreProcessor(new OsuRuleset()) { Mode = { Value = ScoringMode.Standardised } });
+            runForProcessor("lazer-classic", Color4.MediumPurple, new ScoreProcessor(new OsuRuleset()) { Mode = { Value = ScoringMode.Classic } });
+
+            runScoreV1();
+            runScoreV2();
+        }
+
+        private void runScoreV1()
+        {
+            int totalScore = 0;
+            int currentCombo = 0;
+
+            void applyHitV1(int baseScore)
+            {
+                if (baseScore == 0)
+                {
+                    currentCombo = 0;
+                    return;
+                }
+
+                const float score_multiplier = 1;
+
+                totalScore += baseScore;
+
+                // combo multiplier
+                // ReSharper disable once PossibleLossOfFraction
+                totalScore += (int)(Math.Max(0, currentCombo - 1) * (baseScore / 25 * score_multiplier));
+
+                currentCombo++;
+            }
+
+            runForAlgorithm("ScoreV1 (classic)", Color4.Purple,
+                () => applyHitV1(base_great),
+                () => applyHitV1(base_ok),
+                () => applyHitV1(0),
+                () =>
+                {
+                    // Arbitrary value chosen towards the upper range.
+                    const double score_multiplier = 4;
+
+                    return (int)(totalScore * score_multiplier);
+                });
+        }
+
+        private void runScoreV2()
+        {
+            int maxCombo = sliderMaxCombo.Current.Value;
+
+            int currentCombo = 0;
+            double comboPortion = 0;
+            double currentBaseScore = 0;
+            double maxBaseScore = 0;
+            int currentHits = 0;
+
+            for (int i = 0; i < maxCombo; i++)
+                applyHitV2(base_great);
+
+            double comboPortionMax = comboPortion;
+
+            currentCombo = 0;
+            comboPortion = 0;
+            currentBaseScore = 0;
+            maxBaseScore = 0;
+            currentHits = 0;
+
+            void applyHitV2(int baseScore)
+            {
+                maxBaseScore += baseScore;
```

Comment:
minor bug: this should be set to ```base_great``` instead of ``` baseScore``` because in cases of non perfect accuracy will still be at 100%


I noticed this because someone sent me a scorev2 score from stable, and when I tried to graph it, the scoring values didn't add up.

Reply:
Well spotted, have fixed.